File: apps/backend-fastapi/app/schemas/prices.py
# app/schemas/prices.py
from pydantic import BaseModel, Field
from datetime import datetime
from uuid import UUID


class PriceSignalIn(BaseModel):
    market_id: str = Field(..., description="Market UUID")
    size_band_id: str = Field(..., description="Size band UUID")
    price_per_kg: float = Field(..., gt=0)


class ActivePriceAgreementOut(BaseModel):
    market_id: UUID
    size_band_id: UUID
    reference_price: float
    confidence_score: float
    sample_count: int
    valid_from: datetime
    valid_until: datetime


class PriceLockOut(BaseModel):
    id: UUID
    status: str

File: apps/frontend-mobile/api/prices.ts

import { apiRequest } from "./client";

/* =========================
   Types
========================= */

export type ActivePriceAgreement = {
  market_id: string;
  size_band_id: string;
  reference_price: number;
  confidence_score: number;
  sample_count: number;
  valid_from: string;   // ISO datetime
  valid_until: string; // ISO datetime
};



export type ActivePrice = {
  market_id: string
  size_band_id: string
  reference_price: number
  confidence_score: number
  sample_count: number
  valid_from: string
  valid_until: string
}

/* =========================
   Queries
========================= */

export function fetchActivePrices() {
  return apiRequest<ActivePriceAgreement[]>(
    "/prices/active"
  );
}

/* =========================
   Admin actions
========================= */

export function lockPriceAgreement(priceId: string) {
  return apiRequest<{ id: string; status: string }>(
    `/admin/prices/${priceId}/lock`,
    { method: "POST" }
  );
}
